### PR TITLE
Add BaseFileLoader.save()

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

### Added
- `BaseFileLoader.save()` method for saving an Artifact to a destination.
## Issue ticket number and link
Closes https://github.com/griptape-ai/griptape/issues/1429

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - `PromptTask.conversation_memory` for setting the Conversation Memory on a Prompt Task.
 - `Structure.conversation_memory_strategy` for setting whether Conversation Memory Runs should be created on a per-Structure or per-Task basis. Default is `Structure.ConversationMemoryStrategy.PER_STRUCTURE`.
+- `BaseFileLoader.save()` method for saving an Artifact to a destination.
```
Comment (dylanholmes) [inline CHANGELOG.md:14]: Wanna add an example in the docs too? Maybe in the same docs for using the load method?

File: griptape/loaders/base_file_loader.py [hunk 2]
```diff
@@ -27,3 +27,9 @@ def fetch(self, source: str | PathLike) -> bytes:
             return data.encode(self.encoding)
         else:
             return data
+
+    def save(self, destination: str | PathLike, artifact: A) -> None:
```
Comment (dylanholmes) [inline griptape/loaders/base_file_loader.py:31]: Doesn't `PathLike` already include `str`?

File: griptape/loaders/base_file_loader.py [hunk 3]
```diff
@@ -27,3 +27,9 @@ def fetch(self, source: str | PathLike) -> bytes:
...
+        """Saves the Artifact to a destination."""
+        data = artifact.value if isinstance(artifact.value, bytes) else artifact.value.encode(self.encoding)
+
+        self.file_manager_driver.save_file(str(destination), data)
```
Comment (dylanholmes) [inline griptape/loaders/base_file_loader.py:35]: Surprised `save_file` doesn't take `PathLike` too.

**Review by dylanholmes — CHANGES_REQUESTED** [review]
Minor comments

File: griptape/loaders/base_file_loader.py [hunk 4]
```diff
@@ -27,3 +27,9 @@ def fetch(self, source: str | PathLike) -> bytes:
             return data.encode(self.encoding)
         else:
             return data
+
+    def save(self, destination: str | PathLike, artifact: A) -> None:
+        """Saves the Artifact to a destination."""
+        data = artifact.value if isinstance(artifact.value, bytes) else artifact.value.encode(self.encoding)
```
Comment (dylanholmes) [inline griptape/loaders/base_file_loader.py:33]: This is a pretty general serialization operations, seems it would be good in a `to_bytes` method.

Comment (collindutter) [reply griptape/loaders/base_file_loader.py:33]: That reminds me I already added one! Updated.

File: griptape/loaders/base_file_loader.py (hunk 3 above)
Comment (collindutter) [reply griptape/loaders/base_file_loader.py]: Probably should be. Would prefer to keep that change out out of this PR though.

File: CHANGELOG.md (hunk 1 above)
Comment (collindutter) [reply CHANGELOG.md:18]: Added example

File: docs/griptape-framework/data/src/loaders_save.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from griptape.drivers import LocalFileManagerDriver
+from griptape.loaders import TextLoader
+
+loader = TextLoader(file_manager_driver=LocalFileManagerDriver())
+
+data = loader.load("tests/resources/test.txt")
+
+data.value = data.value.replace("foo", "bar")
+
+loader.save("tests/resources/test.txt", data)
```
Comment (dylanholmes) [inline docs/griptape-framework/data/src/loaders_save.py:10]: This is starting to remind me of pathlib

**Review by dylanholmes — APPROVED** [review]

File: docs/griptape-framework/data/src/loaders_save.py (hunk 5 above)
Comment (collindutter) [reply docs/griptape-framework/data/src/loaders_save.py:10]: 🤷

**Pull request merged**